WordPress/gutenberg
# Fluid typography: add configurable settings for minimum font size to theme.json

## What?
Part of 
- https://github.com/WordPress/gutenberg/issues/44888

Follows on from https://github.com/WordPress/gutenberg/pull/39529 by adding configurable fluid font-size settings.

The first setting to be added is `settings.typography.fluid.minFontSize`, which sets the lower bound at which custom font sizes are converted to clamp values.

The current, hard-coded value is `14px`.

## Why?

The hard-coded value of `14px` may not be appropriate for some theme designs.

## How?
By checking global fluid typography for values.

## Testing Instructions

Play with `minFontSize` in the fluid config settings (px/rem/em units only) and ensure that custom font sizes are only clamp above the value you set. 

```json
{
	"version": 2,
	"settings": {
		"appearanceTools": true,
		"layout": {
			"contentSize": "840px",
			"wideSize": "1100px"
		},
		"typography": {
			"fluid": {
				"minFontSize": "16px"
			}
		}
	}
}

```

<details>
<summary>sample editor code</summary>

```html

<h1 class="wp-block-heading" style="font-size:145px">Heading 1</h1>



<p style="font-size:35px">Start with the basic building block of all narrative. Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.Start with the basic building block of all narrative.</p>



```
</details>

Run the tests!

```
npm run test:unit:php -- --filter WP_Block_Supports_Typography_Test
```

```
npm run test:unit packages/block-editor/src/components/font-sizes/test/fluid-utils.js
```

```
npm run test:unit packages/edit-site/src/components/global-styles/test/typography-utils.js
```

```
npm run test:unit packages/block-editor/src/hooks/test/use-typography-props.js
```

## See also

- https://github.com/WordPress/gutenberg/issues/34345
- https://github.com/WordPress/gutenberg/issues/24480
- https://github.com/WordPress/gutenberg/pull/39529
